dil/src/main/java/org/assimbly/dil/blocks/templates/Archive.java

```diff
@@ -0,0 +1,31 @@
+package org.assimbly.dil.blocks.templates;
+
+import org.apache.camel.builder.RouteBuilder;
+
+public class Archive extends RouteBuilder {
+
+     @Override
+     public void configure() throws Exception {
+
+         routeTemplate("archive-action")
+                 .templateParameter("routeconfiguration_id","0")
+                 .templateOptionalParameter("usingIterator", "true")
```

Comment:
I think you mean here:

.templateParameter("usingIterator", "true")

instead of:

.templateOptionalParameter("usingIterator", "true")

With the templateoptionalParameter the second parameter (in this case "true") is only the description, while with the normal templateparameter the second parameter means the "defaultvalue".  Now when the parameter is missing, nothing will happen.